libs/libbuiltin/libgcc/gcov.c

```diff
@@ -0,0 +1,290 @@
+  uint32_t size = sizeof(uint32_t) * 3;
+#endif
+
+  for (iter = 0; iter < info->n_functions; iter++)
+    {
+      size += (sizeof(uint32_t) * 5);
+      for (iter_1 = 0; iter_1 < GCOV_COUNTERS; iter_1++)
+        {
+          if (!info->merge[iter_1])
+            {
+              continue;
+            }
+
+          size += (sizeof(uint32_t) * 2);
+          for (iter_2 = 0; iter_2 < info->functions[iter]->ctrs->num;
+               iter_2++)
+            {
+              size += (sizeof(uint64_t));
+            }
+        }
+    }
+
+  return size;
+}
+
+static size_t gcov_convert(FAR uint8_t *buffer,
+                           FAR const struct gcov_info *info)
+{
+  FAR struct gcov_fn_info *gfi;
+  FAR struct gcov_ctr_info *gci;
+  uint32_t functions;
+  uint32_t counts;
+  uint32_t value;
+  size_t pos = 0;
+
+  dump_unsigned(buffer, &pos, GCOV_DATA_MAGIC);
+  dump_unsigned(buffer, &pos, info->version);
+  dump_unsigned(buffer, &pos, info->stamp);
+
+#ifdef GCOV_12_FORMAT
+  dump_unsigned(buffer, &pos, info->checksum);
+#endif
+
+  for (functions = 0; functions < info->n_functions; functions++)
+    {
+      gfi = info->functions[functions];
+      dump_unsigned(buffer, &pos, GCOV_TAG_FUNCTION);
+      dump_unsigned(buffer, &pos, GCOV_TAG_FUNCTION_LENGTH);
+      dump_unsigned(buffer, &pos, gfi->ident);
+      dump_unsigned(buffer, &pos, gfi->lineno_checksum);
+      dump_unsigned(buffer, &pos, gfi->cfg_checksum);
+
+      gci = gfi->ctrs;
+      for (counts = 0; counts < GCOV_COUNTERS; counts++)
+        {
+          if (!info->merge[counts])
+            {
+              continue;
+            }
+
+          dump_unsigned(buffer, &pos, GCOV_TAG_FOR_COUNTER(counts));
+
+#ifdef GCOV_12_FORMAT
+          dump_unsigned(buffer, &pos, gci->num * 2 * 4);
+#else
+          dump_unsigned(buffer, &pos, gci->num * 2);
+#endif
+
+          for (value = 0; value < gci->num; value++)
+            {
+              dump_counter(buffer, &pos, gci->values[value]);
+            }
+
+          gci++;
+        }
+    }
+
+  return pos;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+void __gcov_init(FAR struct gcov_info *info)
+{
+  info->next = __gcov_info_start;
+  __gcov_info_start = info;
+}
+
+void __gcov_merge_add(FAR gcov_type *counters, unsigned int n_counters)
+{
+}
+
+void __gcov_exit(void)
+{
+}
+
+void __gcov_dump(void)
+{
+}
+
+void __gcov_reset(void)
+{
+  FAR struct gcov_info *info = NULL;
+  FAR struct gcov_fn_info *gfi;
+  FAR struct gcov_ctr_info *gci;
+  uint32_t functions;
+  uint32_t counts;
+  uint32_t values;
+
+  for (info = __gcov_info_start; info; info = info->next)
+    {
+      for (functions = 0; functions < info->n_functions; functions++)
+        {
+          gfi = info->functions[functions];
+          gci = gfi->ctrs;
+
+          for (counts = 0; counts < GCOV_COUNTERS; counts++)
```

Comment:
```suggestion
```

Reply:
The code has been modified. The for loop in the original code is not effective. 
The latest code requires a for loop to traverse gcov_ctr_info and clear the data.

Reference Links
https://github.com/torvalds/linux/blob/master/kernel/gcov/gcc_4_7.c#L218